### PR TITLE
Update Python template.

### PR DESCRIPTION
**Pull request by krivard**
### Description
Include sample code for logging, signal variations, and CSV export.
Update reviewer guide with Make commands for common checks.

### Changelog
Itemize code/test/documentation changes and files added/removed.
- run.py, constants.py - sample code
- setup.py - we now require Python 3.8
- REVIEW.md - use Make

### CODE REVIEW

File: _template_python/delphi_NAME/run.py [hunk 1]
```diff
@@ -25,11 +32,46 @@ def run_module(params):
...
+    mapper = GeoMapper()
+    stats = []
+    ## build the base version of the signal at the most detailed geo level you can get.
+    ## compute stuff here or farm out to another function or file
+    all_data = pd.DataFrame(columns=["timestamp", "val", "zip", "sample_size", "se"])
+    ## aggregate & smooth
+    ## TODO: add num/prop variations if needed
+    for sensor, smoother, geo in product(SIGNALS, SMOOTHERS, GEOS):
+        df = mapper.replace_geocode(
+            all_data, "zip",
```
Comment (chinandrew) [inline _template_python/delphi_NAME/run.py:49]: missing `to_code` (presumably it's `geo`?)
```
def replace_geocode(
        self,
        df,
        from_code,
        new_code,
        from_col=None,
        new_col=None,
        date_col="date",
        data_cols=None,
        dropna=True,
    ):
```

File: _template_python/delphi_NAME/run.py [hunk 2]
```diff
@@ -25,11 +32,46 @@ def run_module(params):
...
     """
-    wip_signal = params["indicator"]["wip_signal"]
-    signal_names = add_prefix(SIGNALS, wip_signal, prefix="wip_")
-    print(signal_names)
+    start_time = time.time()
+    logger = get_structured_logger(
+        __name__, filename=params["common"].get("log_filename"),
+        log_exceptions=params["common"].get("log_exceptions", True))
+    mapper = GeoMapper()
+    stats = []
```
Comment (chinandrew) [inline _template_python/delphi_NAME/run.py:41]: low pri, but might want to consider renaming this to `run_stats` or something else which makes it clear that it's related to the job and not some statistics for the signal

**Comment by chinandrew** [comment]
Should the params json be updated as well?

**Comment by krivard** [comment]
@qx-teo would you check that the default validation params in `params.json.template` look reasonable?

**Review by qx-teo — APPROVED** [review]
validator params look good!

**Review by chinandrew — APPROVED** [review]
:shipit::shipit::shipit:

**Pull request merged**